### ISSUE TITLE
feat(plcs): MotionBricks型VQ tokenを含むcanonical pose出力表現を比較・実装する

### ISSUE THREAD

**Issue by Motoki0705** [Motoki0705/tennis-lab#746]
## 背景

現在のPLCSは、`position`・`rotation`・`presence`を連続値として出力し、canonical 3D poseを有効化する構成ではCOCO17の座標 `[17, 3]` をMLP headから直接回帰する。

一方、MotionBricksはroot軌道を連続値として扱いつつ、身体モーションについてはVQ codebook上のtoken logitを出力し、token選択後にdecoderで連続モーションへ復元する。この方式は、2D観測から一意に定まらない3D poseを単純平均して不自然な姿勢を出す問題や、時間方向のjitterを抑えられる可能性がある。

ただし、MotionBricksのG1 checkpointやロボット骨格をPLCSへ直接移植するのではなく、**PLCSのAMASS/SMPL-H由来canonical COCO17教師を用いて、PLCS向けの出力表現を複数比較する**。

## 目的

PLCSと相性のよいcanonical 3D pose出力表現を複数実装し、同一条件で実学習・評価した上で、最終的に採用すべき方式を根拠付きで提出する。

本Issueの完了条件は設計提案だけではない。候補の実装、学習、評価、可視化、`knowledge/`へのrun記録、最終比較と採否判断までを含む。

## 固定する入出力境界

PLCSの観測入力とtrack-queryの基本出力は変更しない。

```text
input
├─ human_kp / human_vis / detection_mask
├─ court_kp / court_vis
└─ frame / camera / attention masks

root・tracking outputs
├─ position          [B,T,Q,3]
├─ rotation          [B,T,Q,2]  # cos(yaw), sin(yaw)
└─ presence_logits   [B,T,Q]

body output after decode
└─ canonical_pose    [B,T,Q,17,3]
```

単一人物モデルでは`Q`を除いた同等のcontractとする。

`position`・`rotation`・`presence`は現行方式を維持し、比較対象は主にcanonical body poseの出力表現とする。track matchingも当初は現行どおりposition・rotation・presenceで行い、pose関連lossはmatching後に適用する。

## 比較候補

最低限、以下のA〜Dを実装・学習する。EはDの量子化誤差が性能上限になっていることを事前定義した基準で確認した場合のみ追加する。

### A. Direct Cartesian Regression — 現行baseline

```text
hidden state
  -> MLP
  -> canonical_pose [B,T,Q,17,3]
```

現在の`CanonicalPoseHead`相当を比較の基準とする。既存runのうち、現行dataset・座標系・model contractと整合する最良の再現可能runをbaselineとして選ぶ。

### B. Structured Continuous Regression — knowledgeから選定

次の既存候補を中心に`knowledge/`の過去結果を読み、最も比較価値が高い1方式を事前に選定する。

- parent-relative delta
- mean pose + residual
- bone direction / bone-length structured output
- その他、`knowledge/`上でPLCS canonical headとして既に評価された方式

選定理由は学習開始前に記録し、結果を見てから都合のよい方式へ差し替えない。

### C. Distributional Continuous Regression

各座標または構造化要素について点推定だけでなく分布パラメータを出力する。

例:

```text
mu + log_scale
  -> Laplace NLL または Gaussian NLL
  -> post-processはMAP/mean
```

必要性が事前に示された場合のみ小規模mixtureを許可する。無制限なmixture数・複雑なsamplingは導入しない。

### D. MotionBricks-inspired Multi-head VQ Motion Tokens

PLCS専用motion tokenizerをcanonical COCO17系列で学習し、PLCS backboneは離散motion tokenのlogitを出力する。

```text
pose_logits   [B,Q,L,H,K]
  -> argmax / sampling
pose_tokens   [B,Q,L,H]
  -> VQ decoder + root-motion condition
canonical_pose [B,T,Q,17,3]
```

- `L`: motion token length
- `H`: multi-head codebook数
- `K`: headごとのcode数
- 1 tokenが複数frameを表す時間圧縮を用いる
- decoderには必要に応じてroot velocity・yaw velocity・presence maskを条件として与える
- MotionBricksのG1 checkpoint、G1 skeleton、MuJoCo qposは使用しない
- tokenizer/codebookはPLCSのtrain splitのみで学習し、split leakageを禁止する

### E. VQ Decode + Bounded Continuous Residual — 条件付き候補

```text
canonical_pose = VQ_decode(tokens) + bounded_residual
```

Dのoracle token decodeでも再構成誤差が支配的である場合のみ実施する。residualがVQ priorを迂回して通常の直接回帰へ戻らないよう、head容量・ノルム・学習順序を制限する。

## knowledge-firstの学習計画

実装・学習の前に、少なくとも次を読む。

- `knowledge/nodes/group-i530-canonical-head-ablation.md`
- `knowledge/nodes/run-i530-direct-baseline.md`
- `knowledge/nodes/run-i530-parent-delta.md`
- `knowledge/nodes/run-i530-mean-residual.md`
- `knowledge/nodes/run-i530-bone-direction.md`
- その後に追加されたPLCS canonical pose、no-canonical、rotation/velocity、multi-view関連のrun/group node

その上で、比較に使用するbaseline、backbone、dataset artifact、split、loss、optimizer、学習budget、seed、選抜基準を、学習開始前に次の形式でpre-registerする。

```text
knowledge/nodes/group-i<issue>-plcs-output-representation-comparison.md
```

要件:

- arbitraryなdefault configから開始せず、`knowledge/`上の再現可能な比較候補を根拠に選ぶ
- 古い座標変換不具合やschema不整合を含むartifactをbaselineに使用しない
- dataset、split、backbone、augmentation、optimizer、step/epoch budgetを候補間で固定する
- output representation以外の変数を同時に変更しない
- smoothness等の補助lossはknowledge-selected baselineの設定に固定し、本比較中に独立変数として調整しない
- 各runはcommit SHA、完全なcommand/config、dataset version、seed、checkpoint、metrics、可視化、失敗理由を`knowledge/runs/`へ保存する

## 実装方針

候補ごとのif分岐を各PLCSモデルへ散在させず、construction-timeに固定されるtypedなoutput head / decoder contractを導入する。

想定責任:

```text
PLCS backbone
  -> canonical-pose latent/readout
  -> CanonicalPoseOutputHead
  -> raw typed output
  -> task-owned decoder
  -> canonical_pose [..,17,3]
```

要件:

- `forward`およびtransitive helperはtensor計算のみとする
- config schemaは候補ごとにexact validationし、`.get` fallbackやsilent defaultを追加しない
- decode、shape、dtype、mask、座標系はPLCS task-owned model I/O contractが所有する
- single-person、multi-view axial、track-queryで重複実装せず、共通componentを再利用する
- experimental variantは明示configから再実行可能にするが、compatibility alias、pass-through wrapper、到達不能なdead codeは残さない
- prediction TypedDictには有効な候補で必要なraw outputとdecoded `canonical_pose`を明示する
- padding、presence、visibility、label-validityを混同せず、それぞれ別maskとして扱う

### Phase 0: dataset・baseline確定

- [ ] 座標系、SMPL-H→COCO17、canonical/world変換が検証済みのdataset revisionを固定する
- [ ] `knowledge/`から現行baselineを選び、再現runを1本通す
- [ ] 比較指標、seed、選抜基準、失敗条件をpre-registerする

### Phase 1: 表現固有の事前学習

- [ ] D向けVQ tokenizerをtrain splitのみで学習する
- [ ] oracle token decodeの再構成性能、codebook utilization、perplexity、collapse有無を測定する
- [ ] Eへ進む条件をここで判定する

### Phase 2: 全候補のscreening

A〜Dを同一backbone・同一budget・同一seedで学習する。

- [ ] A: direct Cartesian
- [ ] B: knowledge-selected structured continuous
- [ ] C: distributional continuous
- [ ] D: multi-head VQ token
- [ ] E: 条件を満たした場合のみ

### Phase 3: 複数seed比較

screening後、baselineと上位2候補を最低3 seedsで再学習する。上位候補の選抜基準はPhase 0で固定したものを使用する。

### 主指標

- canonical MPJPE
- world-space MPJPE
- position MAE
- yaw angular error
- presence precision / recall / F1

### 構造・時間指標

- bone-length relative error
- joint-angle error
- torsion-angle error
- torso-twist error
- velocity / acceleration / jerk error
- temporal jitter
- track continuityおよび遮蔽区間の誤差

### 分布・token固有指標

- NLLおよびcalibration（C）
- VQ oracle reconstruction error（D/E）
- codebook utilization / perplexity / dead-code率
- token accuracy / top-k accuracy
- quantization error
- deterministic argmaxとsampling時の品質差

### 実用指標

- parameter数
- training/inference peak memory
- inference latency
- decoderを含むthroughput

全体平均だけでなく、motion category、visibility/occlusion、camera数、速度・旋回量などのsliceも提出する。

## 可視化

最低限、同一入力に対して各候補を並べた動画または時系列可視化を提出する。

- GT 3D pose
- direct baseline
- structured/distributional candidate
- VQ candidate
- 代表的成功例
- failure case
- 遮蔽・切り返し・高速移動を含む例

数値指標が改善しても、骨格破綻、foot sliding、姿勢の平均化、token switching、時間的な振動がないことを確認する。

## 成果物

- [ ] 出力head/decoderのtyped共通contract
- [ ] A〜Dの実装とexact config
- [ ] VQ tokenizer、codebook、decoderの学習・推論経路
- [ ] model I/O、loss、matching後supervisionの更新
- [ ] unit / integration / shape / mask / decode / checkpoint tests
- [ ] 実際に完了した学習runとcheckpoint
- [ ] 全候補の比較表
- [ ] 3-seedの統計
- [ ] 可視化artifact
- [ ] `knowledge/runs/`の各run記録
- [ ] `knowledge/nodes/group-i<issue>-plcs-output-representation-comparison.md`の最終まとめ
- [ ] 採用候補、棄却候補、棄却理由、残存リスクの明記

## 完了条件

- 設計・実装だけ、またはtraining queueへの登録だけでは完了としない。実学習結果とartifactを提出すること。
- baselineを含むA〜Dが同一条件で比較されていること。
- baselineと上位2候補について最低3 seedsの結果があること。
- position・rotation・presenceを含む既存PLCS性能が不必要に悪化していないこと。
- winnerをdefaultへ変更する場合は、事前定義した主指標と構造・時間指標の両方で根拠を示すこと。
- 改善しなかった候補もnegative resultとして再現可能に記録すること。改善そのものはIssue closeの必須条件ではないが、比較・分析・採否判断は必須とする。
- 試行錯誤の途中実装をそのまま積み上げず、最終PRでは候補を統一contract下に整理し、config・test・knowledgeから再現可能な状態にすること。

## 対象外

- MotionBricks G1 checkpointの直接利用
- G1→SMPL-H retargeting
- MuJoCoまたは実ロボット制御
- PLCS backbone全体の同時再設計
- position/rotation/presenceまで一括token化する変更
- ラケット・指・SMPL-H全52関節の復元
